esp/services/ws_workunits/ws_workunitsHelpers.hpp

```diff
@@ -333,30 +325,45 @@ struct WUComponentLogOptions
         logFetchOptions.setStartFrom(logFilterReq.getLineStartFrom());
 
-        ILogAccessFilter * componentsFilterObj = nullptr;
         if (logFilterReq.getComponentsFilter().length() > 0)
-            componentsFilterObj = getOredComponentsLogFilter(logFilterReq.getComponentsFilter());
+            extendFetchFilter(logFetchFilter, getOredComponentsLogFilter(logFilterReq.getComponentsFilter()), LOGACCESS_FILTER_and);
 
-        if (componentsFilterObj != nullptr)
-            logFetchFilter.setown(getBinaryLogAccessFilterOwn(logFetchFilter.getLink(), componentsFilterObj, LOGACCESS_FILTER_and));
-
-        ILogAccessFilter * logEventTypeFilterObj = nullptr;
         const char * logType = logFilterReq.getLogEventTypeAsString();
         if (!isEmptyString(logType) && strcmp(logType,"ALL") != 0)
-            logEventTypeFilterObj = getClassLogAccessFilter(LogMsgClassFromAbbrev(logType));
-
-        if (logEventTypeFilterObj != nullptr)
-            logFetchFilter.setown(getBinaryLogAccessFilterOwn(logFetchFilter.getLink(), logEventTypeFilterObj, LOGACCESS_FILTER_and));
+            extendFetchFilter(logFetchFilter, getClassLogAccessFilter(LogMsgClassFromAbbrev(logType)), LOGACCESS_FILTER_and);
 
         const char * wildCharFilter = logFilterReq.getWildcardFilter();
         if (!isEmptyString(wildCharFilter))
-            logFetchFilter.setown(getBinaryLogAccessFilterOwn(logFetchFilter.getLink(), getWildCardLogAccessFilter(wildCharFilter), LOGACCESS_FILTER_or));
+            extendFetchFilter(logFetchFilter, getWildCardLogAccessFilter(wildCharFilter), LOGACCESS_FILTER_or);
 
         logFetchOptions.setFilter(logFetchFilter.getClear());
         CSortDirection espSortDirection = logFilterReq.getSortByTimeDirection();
         logFetchOptions.addSortByCondition(LOGACCESS_MAPPEDFIELD_timestamp, "", espSortDirection == CSortDirection_ASC 
                                         ? SORTBY_DIRECTION_ascending : SORTBY_DIRECTION_descending);
     }
+
+protected:
+    /**
+     * @brief Logically combine a new filter with an existing compound filter.
+     *
+     * Given an existing filter, an extension filter is logically combined with it based on a
+     * requested logical operation. The existing filter is replaced by the combined filter.
+     *
+     * If the existing filter is NULL, it is assigned to the extension. If the extension is NULL,
+     * the existing filter is unchanged.
+     *
+     * @param compoundFilter The existing filter to be logically combined with the extension.
+     * @param extension The new filter to be logically combined with the existing filter.
+     *                  Ownership of the given reference is transferred to the function.
+     * @param operation The logical operation to be performed on the existing and new filters.
+     */
+    void extendFetchFilter(Owned<ILogAccessFilter>& compoundFilter, ILogAccessFilter* extension, LogAccessFilterType operation)
```

Comment:
we should expose this logic at the jlog.hpp level as compoundOwnedFilter(ownedfilter, extention, op)